ngraph_bridge/ngraph_encapsulate_impl.h

```diff
@@ -85,25 +85,43 @@ class NGraphEncapsulateImpl {
       const ng::element::Type& ng_element_type, const ng::Shape& ng_shape,
       std::shared_ptr<ng::runtime::Tensor> tensor_from_pipeline);
 
+  // Clear all maps with ng_exec as keys
+  void ClearExecMaps();
+
+  // Get pipeline index and input and output tensor groups from executable (if
+  // they can create tensors)
+  Status GetPipelineIdxAndTensors(
+      const std::shared_ptr<ngraph::runtime::Executable>&,
+      std::tuple<int, PipelinedTensorVector, PipelinedTensorVector>&);
+
+  // Once done using, return the index to indicate that those executable created
+  // tensors are free for reuse
+  Status ReturnPipelinedTensors(std::shared_ptr<ngraph::runtime::Executable>,
+                                size_t);
+
   void DumpNgFunction(const string&,
                       std::shared_ptr<ngraph::runtime::Executable>);
 
-  // Accessors(getters and setters) for the private data members of
-  // NgraphEncapsulateImpl class
-  // needed by
-  // NgraphEncapsulateOp class
+// Accessors(getters and setters) for the private data members of
+// NgraphEncapsulateImpl class
+// needed by NgraphEncapsulateOp class
+
+#if defined(NGRAPH_TF_ENABLE_VARIABLES_AND_OPTIMIZERS)
   const int& GetNumberOfCopies() { return number_of_copies; }
 
   void SetNumberOfCopies(const int& number) { number_of_copies = number; }
 
-  const int& GetNgraphCluster() { return m_ngraph_cluster; }
-
-  void SetNgraphCluster(const int& cluster) { m_ngraph_cluster = cluster; }
-
   int GetGraphId() { return m_graph_id; }
 
   void SetGraphId(const int& graph_id) { m_graph_id = graph_id; }
 
+  void AppendCopyLog(const string str) { copy_log_str << str; }
```

Comment:
The graph ids are attached irrespective of the build. Need to be out of the ifdef

Reply:
True, but `GetGraphId` is used only in `NGRAPH_TF_ENABLE_VARIABLES_AND_OPTIMIZERS` build

Reply:
Please remove the if-def from the encap_op.cc too. The graph-ids help in debugging/logging and can be used in other builds.  It is also more consistent: we attach it, we read it and keep it handy incase we need it.
We did not have an in-def originally, must have been added somewhere in the refactoring
https://github.com/tensorflow/ngraph-bridge/blob/r0.18/ngraph_bridge/ngraph_encapsulate_op.cc#L110

Reply:
moved graphis functions outside ifdef, and removed ifdef from calling code